test/config.js

```diff
@@ -49,6 +49,15 @@ test.cb('config.fatal = true', t => {
   });
 });
 
+test.cb('config.fatal = true with an exec() failure includes a .code on the Error', t => {
+  shell.config.fatal = true;
+  t.throws(() => {
+    shell.exec('exit 2');
+  }, { code: 2 });
+  shell.config.fatal = false;
```

Comment:
Please move `shell.config.fatal = false` into a `finally` block. I want cleanup code to live in a `finally` block so that it is guaranteed to execute even when the test case fails. Otherwise if this one test case fails, it will leave a side effect which could break every test case which follows this one.